src/arrowc/arrowc.h

```diff
@@ -161,15 +161,82 @@ struct ArrowError {
 /// \brief Represents an errno-compatible error code
 typedef int ArrowErrorCode;
 
-/// }@
-
 /// \brief Set the contents of an error using printf syntax
 ArrowErrorCode ArrowErrorSet(struct ArrowError* error, const char* fmt, ...);
 
 /// \brief Get the contents of an error
 const char* ArrowErrorMessage(struct ArrowError* error);
 
-/// \defgroup arrowc-schema Schema helpers
+/// }@
+
+/// \defgroup arrowc-utils Utility data structures
+
+/// \brief An non-owning view of a string
+struct ArrowStringView {
+  /// \brief A pointer to the start of the string
+  ///
+  /// If n_bytes is 0, this value may be NULL.
+  const char* data;
+
+  /// \brief The size of the string in bytes,
+  ///
+  /// (Not including the null terminator.)
+  int64_t n_bytes;
+};
+
+/// \brief Arrow type enumerator
+enum ArrowType {
```

Comment:
a note from the ADBC review: enum size is not explicitly defined so *if* ABI stability is a concern, using constants instead may be preferable. (If not then an enum is more convenient though.)

Reply:
maybe note explicitly whether this was meant to (not) be convertible to/from `arrow::Type::type`?

Reply:
Are there actual enum size differences between compilers?

Reply:
Ah true. Ok, so this isn't a worry.

I suppose there's no need to comment on the mapping. I don't think we need them to map in any case.

Reply:
Actually, perhaps it would be nice to deliberately making them incompatible so that people don't start relying implicitly on numeric equality.

An easy way to do that is to have `enum ArrowType` start at 1.